rclcpp/include/rclcpp/time_source.hpp

```diff
@@ -140,6 +151,9 @@ class TimeSource
   using SubscriptionT = rclcpp::Subscription<MessageT, Alloc>;
   std::shared_ptr<SubscriptionT> clock_subscription_{nullptr};
   std::mutex clock_sub_lock_;
+  rclcpp::CallbackGroup::SharedPtr clock_callback_group_;
+  rclcpp::executors::SingleThreadedExecutor clock_executor_;
```

Comment:
I think this is the bug. The executor is assuming that the default context was initialized (i.e. `rclcpp::init()` is called).

Reply:
thanks, I see, this is regression. it is expecting default context has been initialized instead core dump. Do we want to revert this fix for now? i think yes, since it's been busy for next release (test, stability and so on) , probably we can come back on this after taking care of this issue?

Reply:
Yeah, I'd say let's revert it for now unless we can fix it quickly.

Reply:
i guess that having clock_executor_ shared_ptr and construct the object based on `node_base_` in `TimeSource`. let me try to give it a shot see if we can fix this quickly.

Reply:
https://github.com/ros2/rclcpp/pull/1617 can fix https://github.com/ros2/rclcpp/pull/1556#issuecomment-812256630